docs/_includes/split-button.html

```diff
@@ -0,0 +1,257 @@
+                    <svg class="icon icon--dropdown" focusable="false" height="10" width="14" aria-hidden="true">
+                        <use xlink:href="#icon-dropdown"></use>
+                    </svg>
+                </button>
+                <div class="menu-button__menu">
+                    <div class="menu-button__items" role="menu">
+                        <div class="menu-button__item" role="menuitem">
+                            <span>Item 10000</span>
+                        </div>
+                        <div class="menu-button__item" role="menuitem">
+                            <span>Item 20000</span>
+                        </div>
+                        <div class="menu-button__item" role="menuitem">
+                            <span>Item 30000</span>
+                        </div>
+                    </div>
+                </div>
+            </span>
+        </div>
+        {% endhighlight %}
+
+        <p>Lots of Text</p>
+
+        <div class="demo">
+            <div class="demo__inner">
+                <div class="split-button">
+                    <button class="btn btn--primary split-button--end btn--truncated" aria-haspopup="true" type="button">
+                        <span class="btn__cell">
+                            <span class="btn__text">Longest Button Text Example Lorem Ipsum Dolor</span>
+                        </span>
+                    </button>
+                        <span class="menu-button">
+                            <button class="btn btn--primary split-button--start" aria-haspopup="true" type="button">
+                                <svg class="icon icon--dropdown" focusable="false" height="10" width="14" aria-hidden="true">
+                                    {% include symbol.html name="dropdown" %}
+                                </svg>
+                            </button>
+                            <div class="menu-button__menu">
+                                <div class="menu-button__items" role="menu">
+                                    <div class="menu-button__item" role="menuitem">
+                                        <span>Item 10000</span>
+                                    </div>
+                                    <div class="menu-button__item" role="menuitem">
+                                        <span>Item 20000</span>
+                                    </div>
+                                    <div class="menu-button__item" role="menuitem">
+                                        <span>Item 30000</span>
+                                    </div>
+                                </div>
+                            </div>
+                        </span>
+                </div>
+            </div>
+        </div>
+        {% highlight html %}
+        <div class="split-button">
+        <button class="btn btn--primary split-button--end" aria-haspopup="true" type="button">
```

Comment:
This is invalid BEM. We have a BEM element modifier here, `split-button--end` but the BEM block the modifier belongs to (`split-button`) exists on a different element. It should always exist on the same element.

We had discussed that this might instead be `btn--split-end`. This would be correct usage of BEM and it would also allow buttons with a non-rounded edge to live outside of the `split-button` class, making it more modular.

Reply:
Resolved